internal/aws/client.go

```diff
@@ -332,6 +333,38 @@ func (c *client) CreateUser(u *User) (user *User, err error) {
 	return
 }
 
+// UpdateUser will update/replace the user specified
+func (c *client) UpdateUser(u *User) (user *User, err error) {
+	startURL, err := url.Parse(c.endpointURL.String())
+	if err != nil {
+		return
+	}
+
+	if u == nil {
+		err = errors.New("no user defined")
+		return
+	}
+
+	startURL.Path = path.Join(startURL.Path, fmt.Sprintf("/Users/%s", u.ID))
+	resp, err := c.sendRequestWithBody(http.MethodPut, startURL.String(), *u)
+	if err != nil {
+		return
+	}
+
+	var newUser User
+	err = json.Unmarshal(resp, &newUser)
+	if err != nil {
+		return
+	}
+	if newUser.ID == "" {
+		user, err = c.FindUserByEmail(u.Username)
+		return
+	}
+
+	user = &newUser
```

Comment:
Could return the value not the reference.

Reply:
Fixed to return the value and not use named return values